lib/tiktoklikescroller.dart

```diff
@@ -2,6 +2,9 @@ library tiktoklikescroller;
 
 import 'package:flutter/widgets.dart';
 
+typedef void ScrollCompleteCallback(int currentIndex);
+typedef void DragBackFirstCallback();
```

Comment:
I'm not quite sure of your specific use-case but to keep things generic it would make sense to have a single callback with maybe several states. I imagine an enum or equivalent to `SCROLLED_FORWARD`, `SCROLLED_BACKWARDS`, `NO_SCROLL_UNDER_THRESHOLD`, `NO_SCROLL_END_OF_LIST`, `NO_SCROLL_START_OF_LIST`. ...or alternatively `currentIndex` and `preScrollIndex`.

Reply:
interesting... I'll check that out on my own app to see how it fits

Reply:
refactored to use one callback